wpengine/faustjs
# feat(useFaustContext.ts): Create and use `useFaustContext` in GQty queries

Signed-off-by: theodesp <theo.despoudis@wpengine.com>

## Description

This PR:
* Adds a new hook `useCheckFaustContext` that will try to resolve the client passed through the `FaustProvider` value and throw an error if it's not provided.
* Fixes a wierd stackoverflow error when `FaustProvider` is missing.
* Forces the use of FaustProvider by calling the `useCheckFaustContext` in all core GQty hooks.
* Adds test case.

## Related Issue(s):



## Testing
* use the Next example folder and remove the <FaustProvider> from the `_app.tsx`. Observe the stack overflow.
* switch to this branch
* use the Next example folder and remove the <FaustProvider> from the `_app.tsx`. Observe the error message.



## Screenshots



**Before when using Faust without a context provider it throws a stack overflow error:**

<img width="1792" alt="Screenshot 2022-02-04 at 10 29 53" src="https://user-images.githubusercontent.com/328805/152515497-db5ca759-f358-482f-93b3-1f56eff75a1d.png">

**After when using Faust without a context provider it throws an proper error message:**
<img width="1791" alt="Screenshot 2022-02-04 at 10 28 05" src="https://user-images.githubusercontent.com/328805/152515672-62d5581d-1287-4e3f-885e-d42a60fdd7af.png">




## Documentation Changes



## Dependant PRs
